data/gui/window/campaign_dialog.cfg

```diff
@@ -195,6 +195,23 @@
 
 								[grid]
 
+									[row]
+										grow_factor = 0
+
+										[column]
+											grow_factor = 1
+											border = "all"
+											border_size = 5
+											horizontal_alignment = "left"
+
+											[text_box]
+												id = "filter_box"
+												definition = "default"
+												{FITER_TEXT_BOX_HINT}
```

Comment:
Typo: "FILTER_TEXT_BOX_HINT".

Reply:
Wow. I implemented the search functionality in the recruit screen and didn't even notice this.

Yes, in this pull request will be fine. Thank you for fixing it. Please make it a separate commit for reviewing and cherry-picking purposes.

Reply:
[Done](https://github.com/wesnoth/wesnoth/pull/3847/commits/dfd9f9f4648898ab4c602a6c8dbf62c7fabd3780)

Reply:
Thank you. I wonder if we should we add a compatibility definition `#define FITER_TEXT_BOX_HINT {FILTER_TEXT_BOX_HINT}#enddef`? Does UMC use this macro?

Reply:
Hm, I don't thought about it. Probably, it's better to add a compatibility definition. But I'm not sure about this :)

Reply:
Let's just add a compatibility definition and move on.

Reply:
Okay, added it.

Reply:
No compatibility macro is needed since this is for internal use only and not exposed to UMC. I've committed (1e00ff8) a fix separately to keep this PR clean.